sql/src/main/java/io/crate/operation/reference/sys/check/cluster/LuceneVersionChecks.java

```diff
@@ -39,4 +45,31 @@ static boolean isUpgradeRequired(@Nullable String versionStr) {
             throw new IllegalArgumentException("'" + versionStr + "' is not a valid Lucene version");
         }
     }
+
+    // We need to check for the version that the index was created since the lucene segments might
+    // be automatically upgraded to the latest version (can happened when data was in the translog) so
+    // minimumCompatVersion cannot be used to indicate that a table recreation is needed.
+    static boolean isRecreationRequired(IndexMetaData indexMetaData) {
+        return indexMetaData != null && indexMetaData.getCreationVersion().before(org.elasticsearch.Version.V_5_0_0);
+    }
+
+    /**
+     * Retrieves an ordered collection of table FQNs that need
+     * to be recreated to be compatible with future CrateDB versions.
+     * @return the ordered collection of table FQNs that need to be recreated
+     */
+    static Collection<String> tablesNeedRecreation(MetaData clusterIndexMetaData) {
+        Collection<String> tablesNeedRecreation = new TreeSet<>();
+        for (ObjectObjectCursor<String, IndexMetaData> entry : clusterIndexMetaData.indices()) {
+            checkIndexMetaData(entry.key, entry.value, tablesNeedRecreation);
```

Comment:
For me it would be more readable to inline the `checkIndexMetaData` in here as it's currently just a method you have to look into to see what it's doing (it's also modifying a collection received as argument and returns `void` so the signature doesn't offer any clue over what it's meant to be doing). 

I think having the `if .. ` statement from `checkIndexMetaData` straight inside this for loop would be easier to read